File: test-automation-core/src/main/java/com/nortal/test/core/file/FileResolver.java
package com.nortal.test.core.file;

import static java.nio.charset.StandardCharsets.UTF_8;

import java.io.IOException;
import java.io.InputStreamReader;
import java.io.Reader;
import java.io.UncheckedIOException;

import lombok.extern.slf4j.Slf4j;
import org.springframework.core.io.Resource;
import org.springframework.core.io.ResourceLoader;
import org.springframework.core.io.support.ResourcePatternUtils;
import org.springframework.stereotype.Component;
import org.springframework.util.FileCopyUtils;

@Slf4j
@Component
public class FileResolver {
	private final ResourceLoader resourceLoader;

	public FileResolver(final ResourceLoader resourceLoader) {
		this.resourceLoader = resourceLoader;
	}

	public String getFileAsString(final String filepath) {
		return asString(getResource(filepath));
	}

	private String asString(Resource resource) {
		try (Reader reader = new InputStreamReader(resource.getInputStream(), UTF_8)) {
			return FileCopyUtils.copyToString(reader);
		} catch (IOException e) {
			throw new UncheckedIOException(e);
		}
	}

	private Resource getResource(final String filepath) {
		return ResourcePatternUtils.getResourcePatternResolver(resourceLoader).getResource(filepath);
	}

}

File: test-automation-core-old/src/main/java/com/nortal/test/core/file/FileResolver.kt
package com.nortal.test.core.file

import lombok.extern.slf4j.Slf4j
import org.springframework.core.io.Resource
import org.springframework.core.io.ResourceLoader
import org.springframework.core.io.support.ResourcePatternUtils
import org.springframework.stereotype.Component
import org.springframework.util.FileCopyUtils
import java.io.IOException
import java.io.InputStreamReader
import java.io.UncheckedIOException
import java.nio.charset.StandardCharsets

@Slf4j
@Component
class FileResolver(private val resourceLoader: ResourceLoader) {
    fun getFileAsString(filepath: String): String {
        return asString(getResource(filepath))
    }

    private fun asString(resource: Resource): String {
        try {
            InputStreamReader(resource.inputStream, StandardCharsets.UTF_8).use { reader -> return FileCopyUtils.copyToString(reader) }
        } catch (e: IOException) {
            throw UncheckedIOException(e)
        }
    }

    private fun getResource(filepath: String): Resource {
        return ResourcePatternUtils.getResourcePatternResolver(resourceLoader).getResource(filepath)
    }
}